### PR TITLE
Add aggregation project stage support

### PR DESCRIPTION
**Pull request by drorasaf**
Add basic projection support, add support for almost all arithmetic opeartors for expression in project, add min operator, add relevant tests

### CODE REVIEW

**Comment by drorasaf** [comment]
Maybe next step is to enable expression parsing for group stage as well

File: mongomock/collection.py [hunk 1]
```diff
@@ -1363,18 +1391,102 @@ def aggregate(self, pipeline, **kwargs):
             '$hour',
             '$minute',
             '$second',
-            '$millisecond']
-        conditional_operators = ['$cond', '$ifNull']  # noqa
+            '$millisecond',
+            '$dateToString']
+
+        def _handle_arithmetic_operator(operator, values, doc_dict):
+            import math
```
Comment (srinivasreddy) [inline mongomock/collection.py]: pls import at the top

File: mongomock/collection.py [hunk 2]
```diff
@@ -1363,18 +1391,102 @@ def aggregate(self, pipeline, **kwargs):
...
+            else:
+                raise NotImplementedError("%s is not supported operator for the aggregation "
+                                          "pipeline. See http://docs.mongodb.org/manual/meta/"
+                                          "aggregation-quick-reference/ for a complete list of "
+                                          "valid operators." % k)
+
+        def _extend_collection(out_collection, field, expression):
+            field_exists = False
+            for doc in out_collection:
+                if field in doc.keys():
```
Comment (srinivasreddy) [inline mongomock/collection.py]: `s/doc.keys()/doc`

File: mongomock/collection.py [hunk 3]
```diff
@@ -1363,18 +1391,102 @@ def aggregate(self, pipeline, **kwargs):
...
+                    field_exists = True
```
Comment (srinivasreddy) [inline mongomock/collection.py]: you need to `break` after field_exists `True`

**Comment by drorasaf** [comment]
@srinivasreddy  Fixed.
Added another fix: instead of getting None return NOTHING, since none might be actual value.

**Comment by srinivasreddy** [comment]
Thanks

**Pull request merged**